### PR TITLE
Feature/6133 regions from settings

### PR DESCRIPTION
**Pull request by jesuspg**
#### Reviewers

@pratid 
#### Description

Use Cache for regions, reading from settings
#### Testing

local environment

### CODE REVIEW

File: dashboard/lib/app.js [hunk 1]
```diff
@@ -82,6 +138,32 @@ function compile(str, path) {
 }
 
 
+function loadRegionsFromSettings() {
```
Comment (jesuspg) [inline dashboard/lib/app.js]: maybe I should add this function to global.regionsCache, something like update or getRegions

File: dashboard/test/unit/test_common.js [hunk 2]
```diff
@@ -21,12 +21,25 @@ var assert = require('assert'),
...
     constants = require('../../lib/constants'),
-    config = require('../../lib/config').data;
+    config = require('../../lib/config').data,
+    _ = require('underscore');
 
 
 /* jshint unused: false */
 suite('common', function () {
 
+    function fillCache(regions) {
```
Comment (jesuspg) [inline dashboard/test/unit/test_common.js:31]: what do you thing about to add this function to global.regionsCache?

File: dashboard/config/settings.json.sample [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        "Region5": {
+            "external_network_name": "public-ext-net-01",
+            "shared_network_name": "node-int-net-01"
+        },
+        "Region6": {
+            "external_network_name": "default",
+            "test_flavor": "tiny"
+        }
+    }
+}
```
Comment (flopezag) [inline dashboard/config/settings.json.sample]: new line

File: dashboard/lib/routes/index.js [hunk 4]
```diff
@@ -133,17 +113,17 @@ function getIndex(req, res) {
...
 
-    cbroker.retrieveAllRegions(txid, function (regions) {
+    cbroker.retrieveAllRegions(txid, function () {
         // Update region sanity_status to `GLOBAL_STATUS_OTHER` for those currently in progress
         jenkins.regionJobsInProgress(txid, function (progress) {
             if (progress) {
-                regions.forEach(function (region) {
-                    if (progress[region.node] === true) {
-                        region.status = constants.GLOBAL_STATUS_OTHER;
+                config.regions.keys.forEach(function (region) {
```
Comment (pratid) [inline dashboard/lib/routes/index.js]: Shouldn't be `keys()`?

File: dashboard/test/unit/test_subscribe.js [hunk 5]
```diff
@@ -130,7 +155,7 @@ suite('subscribe', function () {
 
         //when
         subscribe.isSubscribed(user, region, function () {
-            assert(!region.subscribed);
+//            assert(!region.subscribed);
```
Comment (pratid) [inline dashboard/test/unit/test_subscribe.js]: Is this assert commented on purpose? If so, maybe we should just remove the line

**Comment by pratid** [comment]
LGTM

**Pull request merged**